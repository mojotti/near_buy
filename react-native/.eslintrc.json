{
  "extends": "airbnb",
  "rules": {
    "func-names": ["error", "never"],
    "indent": ["error", 2],
    "arrow-parens": "off",
    "react/jsx-filename-extension": [1, { "extensions": [".js", ".jsx"] }],
    "react/jsx-indent": ["error", 2],
    "react/jsx-indent-props": ["error", 2],
    "import/no-named-as-default": 0,
    "import/prefer-default-export": 0,
    "no-undef": "off",
    "no-use-before-define": "off"
  },
  "env": {
    "jest": true
  },
  "globals": { "fetch": false }
}
