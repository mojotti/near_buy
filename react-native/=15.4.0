NearBuy@0.1.0 /Users/ristomatti/PycharmProjects/near_buy/react-native
└── UNMET PEER DEPENDENCY react@16.0.0

