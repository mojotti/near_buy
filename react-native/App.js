import React from 'react';
import { Provider } from 'react-redux';
import Application from './components/Application';
import store from './redux';

export default class App extends React.Component {
  render() {
    return (
      <Provider store={store}>
        <Application />
      </Provider>
    );
  }
}
