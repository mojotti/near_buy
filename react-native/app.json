{
  "name": "NearBuy",
  "displayName": "NearBuy",
  "expo": {
    "sdkVersion": "25.0.0"
  }
}