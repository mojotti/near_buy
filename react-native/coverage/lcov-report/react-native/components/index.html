<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for react-native/components</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> react-native/components
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">41.57% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>37/89</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">61.54% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>16/26</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">29.79% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>14/47</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">41.57% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>37/89</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="Items.js"><a href="Items.js.html">Items.js</a></td>
	<td data-value="41.18" class="pic low"><div class="chart"><div class="cover-fill" style="width: 41%;"></div><div class="cover-empty" style="width:59%;"></div></div></td>
	<td data-value="41.18" class="pct low">41.18%</td>
	<td data-value="34" class="abs low">14/34</td>
	<td data-value="16.67" class="pct low">16.67%</td>
	<td data-value="6" class="abs low">1/6</td>
	<td data-value="30" class="pct low">30%</td>
	<td data-value="20" class="abs low">6/20</td>
	<td data-value="41.18" class="pct low">41.18%</td>
	<td data-value="34" class="abs low">14/34</td>
	</tr>

<tr>
	<td class="file low" data-value="Login.js"><a href="Login.js.html">Login.js</a></td>
	<td data-value="41.82" class="pic low"><div class="chart"><div class="cover-fill" style="width: 41%;"></div><div class="cover-empty" style="width:59%;"></div></div></td>
	<td data-value="41.82" class="pct low">41.82%</td>
	<td data-value="55" class="abs low">23/55</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="20" class="abs medium">15/20</td>
	<td data-value="29.63" class="pct low">29.63%</td>
	<td data-value="27" class="abs low">8/27</td>
	<td data-value="41.82" class="pct low">41.82%</td>
	<td data-value="55" class="abs low">23/55</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Oct 18 2017 21:32:17 GMT+0300 (EEST)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
