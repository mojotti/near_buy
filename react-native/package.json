{
  "name": "NearBuy",
  "version": "0.1.0",
  "private": true,
  "devDependencies": {
    "@types/js-base64": "^2.1.5",
    "babel-core": "^6.26.3",
    "babel-eslint": "8.2.6",
    "babel-preset-es2015-mod": "^6.6.0",
    "babel-preset-es3": "^1.0.1",
    "babel-preset-react-native-stage-0": "^1.0.1",
    "enzyme": "^3.1.0",
    "enzyme-adapter-react-16": "^1.0.2",
    "enzyme-to-json": "^3.3.4",
    "eslint": "4.19.1",
    "eslint-config-airbnb": "^16.1.0",
    "eslint-plugin-import": "^2.7.0",
    "eslint-plugin-jsx-a11y": "^6.0.2",
    "eslint-plugin-react": "^7.4.0",
    "flow-bin": "^0.49.1",
    "jest": "^23.3",
    "jest-cli": "^21.2.1",
    "prettier": "1.11.0",
    "react-native-mock": "^0.3.1",
    "react-test-renderer": "16.2.0",
    "redux-devtools": "^3.4.0",
    "redux-mock-store": "^1.3.0",
    "regenerator-runtime": "^0.12.1",
    "sinon": "^4.0.1"
  },
  "scripts": {
    "start": "react-native start",
    "android": "react-native run-android",
    "ios": "react-native run-ios",
    "test": "jest",
    "test:watch": "npm test -- --watch",
    "flow": "flow",
    "clean-install": "rm -rf node_modules && rm -rf $TMPDIR/react-* && yarn cache clean --force && watchman watch-del-all && yarn"
  },
  "jest": {
    "testEnvironment": "node",
    "preset": "react-native",
    "setupTestFrameworkScriptFile": "<rootDir>/test-setup.js",
    "transformIgnorePatterns": [
      "node_modules/(?!react-native|react-navigation)/"
    ],
    "setupFiles": [
      "<rootDir>/.config/jest/globalFetch.js"
    ],
    "snapshotSerializers": [
      "enzyme-to-json/serializer"
    ],
    "testPathIgnorePatterns": [
      "/node_modules/",
      "<rootDir>/__tests__/__mocks__/"
    ]
  },
  "dependencies": {
    "babel-jest": "^23.4.2",
    "babel-preset-expo": "^4.0.0",
    "babel-preset-react-native": "5.0.0",
    "base-64": "^0.1.0",
    "isomorphic-fetch": "^2.2.1",
    "lottie-react-native": "^2.5.6",
    "prop-types": "^15.6.1",
    "react": "16.4.1",
    "react-dom": "^16.0.0",
    "react-native": "0.56.0",
    "react-native-fetch-blob": "^0.10.8",
    "react-native-image-crop-picker": "^0.20.3",
    "react-native-image-placeholder": "^1.0.14",
    "react-native-looped-carousel": "^0.1.10",
    "react-native-maps": "^0.21.0",
    "react-native-modal": "^6.5.0",
    "react-native-snap-carousel": "^3.7.1",
    "react-native-swipeout": "^2.3.3",
    "react-native-vector-icons": "^4.5.0",
    "react-navigation": "^1.0.0-beta.13",
    "react-redux": "^5.0.6",
    "redux": "^3.7.2",
    "redux-persist": "^5.9.1",
    "redux-persist-transform-encrypt": "^2.0.0",
    "redux-thunk": "^2.3.0"
  }
}
