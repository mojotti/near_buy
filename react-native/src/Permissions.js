import { PermissionsAndroid } from 'react-native';

export const requestPermissions = () => {
  return requestStoragePermissions()
    .then(granted => {
      // (granted == true) is checked in case of API levels < 23, which
      // does not return 'granted' like never APIs do.
      if (granted === PermissionsAndroid.RESULTS.GRANTED || granted === true) {
        return requestLocationPermissions();
      } else {
        throw new Error('Permissions are not granted');
      }
    })
    .then(granted => {
      // (granted == true) is checked in case of API levels < 23, which
      // does not return 'granted' like newer APIs do.
      if (granted === PermissionsAndroid.RESULTS.GRANTED || granted === true) {
        return requestCameraPermissions();
      } else {
        throw new Error('Permissions are not granted');
      }
    })
    .then(granted => {
      if (granted === PermissionsAndroid.RESULTS.GRANTED || granted === true) {
        return Promise.resolve(granted);
      } else {
        throw new Error('Permissions are not granted');
      }
    });
};

const requestLocationPermissions = () => {
  return PermissionsAndroid.request(
    PermissionsAndroid.PERMISSIONS.ACCESS_FINE_LOCATION,
    {
      title: 'Location Permissions',
      message: 'NearBuy needs to access your location',
    },
  ).then(granted => {
    if (granted === PermissionsAndroid.RESULTS.GRANTED || granted === true) {
      return Promise.resolve(granted);
    } else {
      throw new Error('Location permissions are not granted');
    }
  });
};

const requestStoragePermissions = () => {
  return PermissionsAndroid.request(
    PermissionsAndroid.PERMISSIONS.READ_EXTERNAL_STORAGE,
    PermissionsAndroid.PERMISSIONS.WRITE_EXTERNAL_STORAGE,
    {
      title: 'Storage Permissions',
      message: 'NearBuy needs the access to your external storage',
    },
  ).then(granted => {
    if (granted === PermissionsAndroid.RESULTS.GRANTED || granted === true) {
      return Promise.resolve(granted);
    } else {
      throw new Error('Storage permissions are not granted');
    }
  });
};

const requestCameraPermissions = () => {
  return PermissionsAndroid.request(PermissionsAndroid.PERMISSIONS.CAMERA, {
    title: 'Camera Permissions',
    message: 'NearBuy needs to use camera',
  }).then(granted => {
    if (granted === PermissionsAndroid.RESULTS.GRANTED || granted === true) {
      return Promise.resolve(granted);
    } else {
      throw new Error('Storage permissions are not granted');
    }
  });
};
