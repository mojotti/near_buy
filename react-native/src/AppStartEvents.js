import { getCurrentLocation } from './Location';
import { requestPermissions } from './Permissions';

export const runAppStartEvents = () => {
  requestPermissions().then(() => getCurrentLocation());
};
