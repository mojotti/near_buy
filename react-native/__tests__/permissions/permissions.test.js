import { PermissionsAndroid } from 'react-native';
import { requestPermissions } from '../../src/Permissions';

describe('permission tests', () => {
  it('it does not throw error when permissions are granted', () => {
    const permissionSpy = {
      request: () => {
        return new Promise.resolve(PermissionsAndroid.RESULTS.GRANTED);
      },
    };

    return requestPermissions(permissionSpy).then(result =>
      expect(result).toMatchSnapshot(),
    );
  });

  it('throws error when permissions are not granted', () => {
    const permissionSpy = {
      request: () => {
        return new Promise.resolve(PermissionsAndroid.RESULTS.DENIED);
      },
    };

    return requestPermissions(permissionSpy).catch(error =>
      expect(error).toMatchSnapshot(),
    );
  });
});
