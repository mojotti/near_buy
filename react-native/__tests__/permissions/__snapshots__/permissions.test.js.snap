// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`permission tests it does not throw error when permissions are granted 1`] = `"granted"`;

exports[`permission tests throws error when permissions are not granted 1`] = `[Error: Storage permissions are not granted]`;
